src/main/DependencyUpdate/goDependencyUpdate.ts

```diff
@@ -0,0 +1,21 @@
+import { GoTreeNode } from '../treeDataProviders/dependenciesTree/dependenciesRoot/goTree';
+import { DependenciesTreeNode } from '../treeDataProviders/dependenciesTree/dependenciesTreeNode';
+import { GoUtils } from '../utils/goUtils';
+import { ScanUtils } from '../utils/scanUtils';
+import { AbstractUpdateDependency } from './abstractDependencyUpdate';
+
+export class GoUpdateDependency extends AbstractUpdateDependency {
+    constructor() {
+        super(GoUtils.PKG_TYPE);
+    }
+
+    /** @override */
+    public updateDependencyVersion(dependenciesTreeNode: DependenciesTreeNode, fixedVersion: string): void {
+        if (!(dependenciesTreeNode.parent instanceof GoTreeNode)) {
+            return;
+        }
```

Comment:
Please put all preconditions like this in an overriding `isMatched` function. Same for npm and Maven